internal/engine/ingester/diff/diff.go

```diff
@@ -203,6 +209,124 @@ func (di *Diff) ingestFileForDepDiff(
 	return batchCtxDeps, nil
 }
 
+func (di *Diff) getScalibrTypeDiff(ctx context.Context, _ int, pr *pb.PullRequest) (*engif.Result, error) {
+	deps := pbinternal.PrDependencies{Pr: pr}
+
+	// TODO: we should be able to just fetch the additional commits between base and target.
+	// Our current Git abstraction isn't quite powerful enough, so we do two shallow clones.
+
+	baseInventory, err := di.scalibrInventory(ctx, pr.BaseCloneUrl, pr.BaseRef)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to clone base from %s at %q: %w", pr.BaseCloneUrl, pr.BaseRef, err)
+	}
+	newInventory, err := di.scalibrInventory(ctx, pr.TargetCloneUrl, pr.TargetRef)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to clone fork from %s at %q: %w", pr.TargetCloneUrl, pr.TargetRef, err)
+	}
+
+	newDeps := setDifference(baseInventory, newInventory, inventorySorter)
+
+	deps.Deps = make([]*pbinternal.PrDependencies_ContextualDependency, 0, len(newDeps))
+	for _, inventory := range newDeps {
+		for _, filename := range inventory.Locations {
+			deps.Deps = append(deps.Deps, &pbinternal.PrDependencies_ContextualDependency{
+				Dep: &pbinternal.Dependency{
+					Ecosystem: inventoryToEcosystem(inventory),
+					Name:      inventory.Name,
+					Version:   inventory.Version,
+				},
+				File: &pbinternal.PrDependencies_ContextualDependency_FilePatch{
+					Name:     filename,
+					PatchUrl: "", // TODO: do we need this?
```

Comment:
Not to diff the deps, but we'd need this in case we wanted to provide the inline reviews. (Which in their current form where we just propose the version bump w/o proposing also any possible transitive deps is...demo quality)

Reply:
Yeah, I want to think through what the abstract interface is for proposing patch comments upstream, but I need to take a look at the GitHub PR and GitLab MR comment APIs before I propose something concrete.  (In any case, I think this is a place where we want to be able to leverage the providers so this can be re-used across multiple repo providers.)